terraform/files/template/cluster-template.yaml

```diff
@@ -7,7 +7,9 @@ metadata:
 spec:
   clusterNetwork:
     pods:
-      cidrBlocks: ["192.168.0.0/16"]    # CIDR block used by Calico.
+      cidrBlocks: ["${POD_CIDR}"]    # CIDR block used by Calico.
```

Comment:
This optional value should be propagated also to the selected CNI solution. Currently, the default CNI is Calico, but there is an open [PR ](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/pull/431) which makes Cilium the default CNI. 

AFAIK Calico (somehow) "auto-discover" k8s pod CIDR and  by default, creates an IP pool for the entire Kubernetes pod CIDR see [here](https://docs.tigera.io/calico/latest/networking/ipam/legacy-firewalls#kubernetes-pod-cidr) 

Would be great to test both Calico/Cilium and ensure that optional POD CIDR is adopted by the CNI.

Reply:
/tested 

Both CNI solutions work as expected. See a brief test report:

CALICO: 

```bash
# Create cluster with custom Pod/Service CIDRs
TF_VAR_pod_cidr=172.17.0.0/16 TF_VAR_service_cidr=192.168.0.0/16 OS_CLOUD=test1 make create

# Wait for cluster creation complete, get kubeconfig of created cluster and create a Pod and expose it via Service
kubectl run nginx --image nginx
kubectl expose pod nginx --port=8080 --name=nginx

# Validate Pod/Service CIDRs
$ kubectl get po nginx -o wide
NAME    READY   STATUS    RESTARTS   AGE   IP              NODE                            NOMINATED NODE   READINESS GATES
nginx   1/1     Running   0          75s   172.17.92.136   test1-test1-md-0-genw01-lvwtc   <none>           <none>

$ kubectl get svc nginx 
NAME    TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
nginx   ClusterIP   192.168.3.70   <none>        8080/TCP   100s
```

CILIUM:


```bash
# Create cluster with custom Pod/Service CIDRs
TF_VAR_use_cilium=true TF_VAR_pod_cidr=172.17.0.0/16 TF_VAR_service_cidr=192.168.0.0/16 OS_CLOUD=test1 make create

# Wait for cluster creation complete, get kubeconfig of created cluster and create a Pod and expose it via Service
kubectl run nginx --image nginx
kubectl expose pod nginx --port=8080 --name=nginx

# Validate Pod/Service CIDRs
$ kubectl get po nginx -o wide
NAME    READY   STATUS    RESTARTS   AGE   IP             NODE                            NOMINATED NODE   READINESS GATES
nginx   1/1     Running   0          11s   172.17.1.116   test1-test1-md-0-genw01-8lfpv   <none>           <none>

$ kubectl get svc nginx 
NAME    TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
nginx   ClusterIP   192.168.16.89   <none>        8080/TCP   22s
```

```bash
# Validate Pod CIDR setting in cilium-config 
$ kubectl -n kube-system get cm cilium-config -oyaml | yq .data.cluster-pool-ipv4-cidr
172.17.0.0/16
```